src/vs/workbench/contrib/notebook/browser/notebookExecutionStateServiceImpl.ts

```diff
@@ -22,10 +22,14 @@ export class NotebookExecutionStateService extends Disposable implements INotebo
 	private readonly _executions = new ResourceMap<Map<number, CellExecution>>();
 	private readonly _notebookListeners = new ResourceMap<NotebookExecutionListeners>();
 	private readonly _cellListeners = new ResourceMap<IDisposable>();
+	private readonly _lastFailedCells = new ResourceMap<number>();
```

Comment:
This maps to cell handle (instead of cellview model) because importing `ICellViewModel` in `INotebookExecutionStateService` would require moving `INotebookExecutionStateService` to browser, which in turn would require a whole lot of files to be moved to browser. Seems that this is the most low-impact.